### PR TITLE
Add user configurable embedding model and quota (LlamaIndex on Vertex AI for RAG)

### PR DESCRIPTION
**Pull request by yinghsienwu**
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [ ] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: generative_ai/rag.py [hunk 1]
```diff
@@ -35,7 +35,17 @@ def create_corpus(
     # Initialize Vertex AI API once per session
     vertexai.init(project=project_id, location="us-central1")
 
-    corpus = rag.create_corpus(display_name=display_name, description=description)
+    # Configure embedding model
+    embedding_model_config = rag.EmbeddingModelConfig(
+        publisher_model="publishers/google/models/textembedding-gecko@003"
```
Comment (davenportjw) [inline generative_ai/rag.py:40]: Can we update this to textembedding-004?

File: generative_ai/rag.py [hunk 2]
```diff
@@ -342,20 +354,29 @@ def quickstart(
...
+    # paths = ["https://drive.google.com/file/d/123", "gs://my_bucket/my_files_dir"]  # Supports Google Cloud Storage and Google Drive Links
 
     # Initialize Vertex AI API once per session
     vertexai.init(project=project_id, location="us-central1")
 
     # Create RagCorpus
-    rag_corpus = rag.create_corpus(display_name=display_name)
+    # Configure embedding model, for example "textembedding-gecko@003".
+    embedding_model_config = rag.EmbeddingModelConfig(
+        publisher_model="publishers/google/models/textembedding-gecko@003"
```
Comment (davenportjw) [inline generative_ai/rag.py:365]: Can we update this to textembedding-004?

**Review by davenportjw — CHANGES_REQUESTED** [review]

File: generative_ai/rag.py (hunk 2 above)
Comment (yinghsienwu) [reply generative_ai/rag.py]: done

File: generative_ai/rag.py (hunk 1 above)
Comment (yinghsienwu) [reply generative_ai/rag.py]: done

**Review by holtskinner — APPROVED** [review]

**Review by davenportjw — APPROVED** [review]

**Pull request merged**